internal/querynodev2/services.go

```diff
@@ -253,6 +253,13 @@ func (node *QueryNode) WatchDmChannels(ctx context.Context, req *querypb.WatchDm
 	}
 	defer node.subscribingChannels.Remove(channel.GetChannelName())
 
+	// to avoid concurrent watch/unwatch
+	if node.unsubscribingChannels.Contain(channel.GetChannelName()) {
```

Comment:
how does this work if the unsubscribing is coming but has not modified the `unsubscribingChannels`?